File: tablet/Resources/menu/ClinicalMenu.js
// ClinicalMenu.js

/*
    Copyright (C) 2012-2015 Rudolf Cardinal (rudolf@pobox.com).
    Department of Psychiatry, University of Cambridge.
    Funded by the Wellcome Trust.

    This file is part of CamCOPS.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/*jslint node: true, newcap: true */
"use strict";
/*global L */

module.exports = function ClinicalMenu() {

    var MenuWindow = require('menulib/MenuWindow'),
        UICONSTANTS = require('common/UICONSTANTS'),
        ALLTASKS = require('common/ALLTASKS'),
        data = [
            UICONSTANTS.CHANGE_PATIENT_MENU_LINE,
            ALLTASKS.TASKLIST.BMI,
            ALLTASKS.TASKLIST.CONTACTLOG,
            ALLTASKS.TASKLIST.CPFT_LPS_REFERRAL,
            ALLTASKS.TASKLIST.CPFT_LPS_RESETRESPONSECLOCK,
            ALLTASKS.TASKLIST.CPFT_LPS_DISCHARGE,
            ALLTASKS.TASKLIST.DIAGNOSIS_ICD10,
            ALLTASKS.TASKLIST.FFT,
            ALLTASKS.TASKLIST.IRAC,
            ALLTASKS.TASKLIST.PATIENT_SATISFACTION,
            ALLTASKS.TASKLIST.PHOTO,
            ALLTASKS.TASKLIST.PHOTOSEQUENCE,
            ALLTASKS.TASKLIST.PROGRESSNOTE,
            ALLTASKS.TASKLIST.PSYCHIATRICCLERKING,
            ALLTASKS.TASKLIST.REFERRER_SATISFACTION_SPEC,
        ],
        self = new MenuWindow({
            title: L('menutitle_clinical'),
            icon: UICONSTANTS.ICON_MENU_CLINICAL,
            data: data,
        });

    return self;
};

File: tablet_qt/menu/clinicalmenu.cpp
/*
    Copyright (C) 2012, University of Cambridge, Department of Psychiatry.
    Created by Rudolf Cardinal (rnc1001@cam.ac.uk).

    This file is part of CamCOPS.

    CamCOPS is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    CamCOPS is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CamCOPS. If not, see <https://www.gnu.org/licenses/>.
*/

#include "clinicalmenu.h"

#include "common/uiconst.h"
#include "lib/uifunc.h"
#include "menulib/menuitem.h"
#include "tasks/bmi.h"
#include "tasks/contactlog.h"
#include "tasks/cpftlpsdischarge.h"
#include "tasks/cpftlpsreferral.h"
#include "tasks/cpftlpsresetresponseclock.h"
#include "tasks/diagnosisicd10.h"
#include "tasks/gbogpc.h"
#include "tasks/gbogras.h"
#include "tasks/gbogres.h"
#include "tasks/irac.h"
#include "tasks/photo.h"
#include "tasks/photosequence.h"
#include "tasks/progressnote.h"
#include "tasks/psychiatricclerking.h"

ClinicalMenu::ClinicalMenu(CamcopsApp& app) :
    MenuWindow(app, uifunc::iconFilename(uiconst::ICON_CLINICAL))
{
}

QString ClinicalMenu::title() const
{
    return tr("Clinical notes and logs");
}

void ClinicalMenu::makeItems()
{
    m_items = {
        MAKE_CHANGE_PATIENT(m_app),
        MAKE_TASK_MENU_ITEM(Bmi::BMI_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(ContactLog::CONTACTLOG_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(CPFTLPSReferral::CPFTLPSREFERRAL_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(
            CPFTLPSResetResponseClock::CPFTLPSRESETCLOCK_TABLENAME, m_app
        ),
        MAKE_TASK_MENU_ITEM(
            CPFTLPSDischarge::CPFTLPSDISCHARGE_TABLENAME, m_app
        ),
        MAKE_TASK_MENU_ITEM(DiagnosisIcd10::DIAGNOSIS_ICD10_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(GboGReS::GBOGRES_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(GboGPC::GBOGPC_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(GboGRaS::GBOGRAS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Irac::IRAC_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Photo::PHOTO_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(PhotoSequence::PHOTOSEQUENCE_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(ProgressNote::PROGNOTE_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(PsychiatricClerking::PSYCLERK_TABLENAME, m_app),
    };
}
